CMakeLists.txt

```diff
@@ -195,6 +193,9 @@ add_compile_options(-Wall)
 add_compile_options(-Werror)
 add_compile_options(-Wunused-parameter)
 add_compile_options(-Wshadow)
+# Disable the complain anout comparison between signed and unsigned integers
+# TODO (sye) Need to figure out a better solution
+add_compile_options(-Wno-error=sign-compare)
```

Comment:
We should decide whether to add this option based on the compiler version.

Reply:
I prefer not to add this option at the global level at all. We can fix all warnings in our code. The trouble is caused by folly. This is just a temporary solution. That's why I added a TODO here